### ISSUE TITLE
Add support for gain maps

### ISSUE THREAD

**Issue by vorg** [pex-gl/pex-renderer#374]
https://gainmap-creator.monogrid.com

200MB exr -> 10MB jpg? Hell yeah!

Questions:
- [ ] is gain map support pex-renderer, pex-shaders or lib-renderer feature?
- [ ] can i resize the jpg and keep metadata? Does it still work? E.g. That 10MB jpg is 8k. Can i resize ldr jpg to 1024x512 and still get correct hdr? Resizing orignal hdr/exr is tricky without Photohop.

**Comment by dmnsgn** [comment]
Summary:
- a [pex-loaders](https://github.com/pex-gl/pex-loaders/blob/5fbd31a3a72c7a3081b406a078b56708593f60de/ultra-hdr.js#L251) feature, without any dependency other than a pex-context to recover the gainmap data
- uses [SRGB](https://developer.mozilla.org/en-US/docs/Web/API/EXT_sRGB) texture so WebGL1 support not guaranteed (but we're sunsetting it soon anyway)
- resizing workflow without third party need clarifications
- jpeg only for now
- jpeg quality when exporting also yields further size decrease (huge drop in size from quality 1 to 0.9, possibly in the 5MB to 500kb) with little compromise on the env map quality.
- I'd recommend we also agree on hdr file naming, including jpeg quality in the file naming: `source-name-size-quality.ext` eg. `polyhaven-brown_photostudio_02-4096x2048-q90.jpg`

Notes:

```bash
# Get jpeg image quality
identify -format '%Q' file.jpg
```

**Comment by vorg** [comment]
>  can i resize the jpg and keep metadata
No. You need to resize .hdr e.g. in Pixelmator and generate new gaim map
